### ISSUE TITLE
Leading underscore not trimmed in object keys

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#328]
Forwarding from https://github.com/prettier/prettier/issues/2143.

Input:
```ts
obj = {                                                                               
  '__': 42 
}
```

Output:
```ts
obj = {                                                                               
  '___': 42 
}
```

Looks the same as https://github.com/eslint/typescript-eslint-parser/issues/296

**Comment by soda0289** [comment]
Thanks for the report. I wonder if there are any other places this happens. Pretty much anywhere we have an identifier, that's not an identifier node type, we need to escape the value.

**Comment by azz** [comment]
Curious why TS needs to "escape" it in the first place?

**Comment by JamesHenry** [comment]
@azz This is the original issue we reported to them https://github.com/Microsoft/TypeScript/issues/14013, you can see their response on there

**Comment by azz** [comment]
Thanks! Seems like it could have been avoided with a good old `hasOwnProperty` check? 😉
